src/odemis/driver/xt_client.py

```diff
@@ -2271,6 +2281,55 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
                         moving = self.parent.stage_is_moving()
                         if moving:
                             logging.warning("Stage reported stopped but moving again, will wait longer")
+                            continue
+
+                    # The stage is not moving anymore, but we still want to wait until the position has been updated
+                    check_pos = True
+                    # check for rotation and tilt is not included as it was not tested on the hardware
+                    # do not check update if angular movements are present
+                    if 'r' in pos or 't' in pos:
+                        logging.debug("Position includes rx or rz. Breaking out of the loop.")
+                        check_pos = False
+                    else:
+                        # only linear movements are requested, estimate time taken by stage
+                        # TODO check the linear speed, default to 2 (copied from move_to_pos in zeiss.py)
+                        pass
+                        # dur_lin = max(
+                        #     numpy.sqrt((pos - orig_pos_raw) ** 2)) / 2
+                        # # give enough time for the stage to update its position
+                        # timeout_update = dur_lin*10
+                    timeout_update = 10  # keep it to 10 seconds till the linear speed of stage is verified
+                    time_check = time.time()
+                    while check_pos:
+                        if time.time() > timeout_update + time_check:
+                            logging.warning(
+                                "Stage position after move + %s s is still not properly updated from original pos: "
+                                "%s. Giving up waiting", timeout_update, orig_pos_raw)
+
```

Comment:
I think you need a break here to exit
```suggestion
                            logging.warning(
                                "Stage position after move + %s s is still not properly updated from original pos: "
                                "%s. Giving up waiting", timeout_update, orig_pos_raw)
                            break

```